cli/asc.js

```diff
@@ -212,15 +212,24 @@ exports.main = function main(argv, options, callback) {
   // Set up transforms
   const transforms = [];
   if (args.transform) {
-    args.transform.forEach(transform =>
-      transforms.push(
-        require(
-          path.isAbsolute(transform = transform.trim())
-            ? transform
-            : path.join(process.cwd(), transform)
-        )
-      )
-    );
+    args.transform.forEach(filename => {
+      filename = path.isAbsolute(filename = filename.trim())
+        ? filename
+        : path.join(process.cwd(), filename);
+      if (/\.ts$/.test(filename)) require("ts-node").register({ transpileOnly: true, skipProject: true });
+      const classOrModule = require(filename);
+      if (typeof classOrModule === "function") {
+        Object.defineProperties(classOrModule.prototype, {
```

Comment:
How about replace this to simpler:
```ts
Object.assign(classOrModule.prototype, {
  baseDir,
  readFile,
  writeFile,
  listFiles
});
```

Reply:
You could use `Object.freeze(Object.assign( ... ))` if you care about immutability.